addons/l10n_it_edi/models/account_move.py

```diff
@@ -1031,117 +1029,97 @@ def _compose_info_message(self, tree, tags):
-            errors.append(_("%s must have a country.", seller.display_name))
-
-        if not is_self_invoice and seller.l10n_it_has_tax_representative and not seller.l10n_it_tax_representative_partner_id.vat:
-            errors.append(_("Tax representative partner %s of %s must have a tax number.", seller.l10n_it_tax_representative_partner_id.display_name, seller.display_name))
-
-        # <1.4.1>
-        if not buyer.vat and not buyer.l10n_it_codice_fiscale and buyer.country_id.code == 'IT':
-            errors.append(_("The buyer, %s, or his company must have a VAT number and/or a tax code (Codice Fiscale).", buyer.display_name))
-
-        if is_self_invoice and self._l10n_it_edi_services_or_goods() == 'both':
-            errors.append(_("Cannot apply Reverse Charge to a bill which contains both services and goods."))
-
-        if is_self_invoice and not buyer.partner_id.l10n_it_pa_index:
-            errors.append(_("Vendor bills sent as self-invoices to the SdI require a valid PA Index (Codice Destinatario) on the company's contact."))
-
-        for tax_line in self.line_ids.filtered(lambda line: line.tax_line_id):
-            if not tax_line.tax_line_id.l10n_it_exempt_reason and tax_line.tax_line_id.amount == 0:
-                errors.append(_("%s has an amount of 0.0, you must indicate the kind of exoneration.", tax_line.name))
-
-        if self.l10n_it_partner_pa:
-            if not self.l10n_it_origin_document_type:
-                errors.append(_("This invoice targets the Public Administration, please fill out"
-                                " Origin Document Type field in the Electronic Invoicing tab."))
-            if self.l10n_it_origin_document_date and self.l10n_it_origin_document_date > fields.Date.today():
-                errors.append(_("The Origin Document Date cannot be in the future."))
-
-        errors += self._l10n_it_edi_export_taxes_data_check()
+        # Fetch errors from company, partner, representative
+        for move in self:
+            for key, error in move.company_id._l10n_it_edi_export_check().items():
+                record = self.env['res.config.settings'] if key.startswith("settings_") else move.company_id
+                save_error(record, error, key)
+            partner_kind = 'other' if not move._l10n_it_edi_is_simplified() else 'simplified'
+            for key, error in move.commercial_partner_id._l10n_it_edi_export_check(partner_kind).items():
+                save_error(move.commercial_partner_id, error, key)
+            for key, error in move.company_id.l10n_it_tax_representative_partner_id._l10n_it_edi_export_check('representative'):
+                save_error(move.company_id, error, key)
+
+        errors = {}
+        for key_without_model, models_dict in error_buffer.items():
+            # Companies partners may appear twice in case they have different roles in the moves.
+            # We merge companies warnings into the partner ones for consistency
```

Comment:
This can not be solved by checking above if they are different instead of this merge?